earn/src/components/boost/ImportModal.tsx

```diff
@@ -175,77 +259,147 @@ export default function ImportModal(props: ImportModalProps) {
     confirmations: 1,
     hash: mintTxn?.hash,
     chainId: activeChain.id,
-    onSuccess(data) {
-      // TODO: prob want to reload the page in here. keep the console.debug if possible (for consistency since we have this style of thing elsewhere)
-      console.debug('Mint transaction successful!', data);
+    onSuccess() {
+      navigate(0);
     },
   });
 
-  // TODO: you can remove these console.logs obviously
-  let state: ImportModalState;
+  let state: ImportModalState | undefined;
   if (isFetchingManager) {
     state = ImportModalState.FETCHING_DATA;
-    console.log('fetching');
   } else if (isWritingManager || isMinting) {
     state = ImportModalState.WAITING_FOR_TRANSACTION;
-    console.log('waiting for txn');
   } else if (isAskingUserToWriteManager) {
     state = ImportModalState.ASKING_USER_TO_APPROVE;
-    console.log('asking user to approve');
   } else if (isAskingUserToMint) {
     state = ImportModalState.ASKING_USER_TO_MINT;
-    console.log('asking user to mint');
   } else if (shouldWriteManager && writeManager) {
     state = ImportModalState.READY_TO_APPROVE;
-    console.log('ready to approve');
   } else if (shouldMint && mint) {
     state = ImportModalState.READY_TO_MINT;
-    console.log('ready to mint');
   }
 
+  const buttonState = getButtonState(state);
+
   // Generate labels for input range (slider)
   const labels: string[] = [];
   for (let i = BOOST_MIN; i <= BOOST_MAX; i += 1) {
-    labels.push(`${i.toFixed(0)}x`);
+    if (i % 2 !== 0) {
+      labels.push(`${i.toFixed(0)}x`);
+    } else {
+      labels.push('');
+    }
   }
 
+  const updatedCardInfo: BoostCardInfo | undefined = useMemo(() => {
+    if (!cardInfo) return undefined;
+    const { position } = cardInfo;
+    const updatedLiquidity = JSBI.multiply(position.liquidity, JSBI.BigInt(boostFactor));
+    return new BoostCardInfo(
```

Comment:
You could write a static factory method on `BoostCardInfo` that allows for usage like:

```
return BoostCardInfo.from(cardInfo, {
  marginAccount: {...},
  position: {...},
})
```

Reply:
Punting